pkg/models/monitoring/sort_page.go

```diff
@@ -50,9 +50,22 @@ func (w wrapper) Less(i, j int) bool {
 	p := w.MetricValues[i]
 	q := w.MetricValues[j]
 
-	if p.Sample.Value() == q.Sample.Value() {
+	// Place Nil to the tail.
+	if p.Sample == nil && q.Sample != nil {
+		return false
+	}
+	if p.Sample != nil && q.Sample == nil {
+		return true
+	}
+
+	// If both Samples are Nil or have the same metric value, sort by resource name
+	if p.Sample == q.Sample || p.Sample[1] == q.Sample[1] {
 		return p.Metadata[w.identifier] < q.Metadata[w.identifier]
 	}
+	// Place NaN to the tail (NaN takes precedence over Nil).
+	if math.IsNaN(p.Sample[1]) != math.IsNaN(q.Sample[1]) {
+		return !math.IsNaN(p.Sample[1])
```

Comment:
NaN value is placed after non-NaN value no matter it's ascending or descending order?

Reply:
Yep, again, because we have well-designed builtin promqls. NaN (and +Inf, -Inf) kindof never occurs. But it's possible in custom monitoring.

See test data:
https://github.com/kubesphere/kubesphere/blob/14069b95f231440a9254b8ec5c700d397d90adf3/pkg/models/monitoring/testdata/faulty-node-metrics-sorted-3.json#L5-L81